OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
         }
 
         #endregion
+
+        #region Test GetFeatureVariable<Type> Typecasting
+
+        [Test]
+        public void TestGetFeatureVariableBooleanReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyTrue = "varTrue";
+            var variableKeyFalse = "varFalse";
+            var variableKeyNonBoolean = "varNonBoolean";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.BOOLEAN;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyTrue, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("True");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyFalse, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("false");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNonBoolean, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_boolean_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
```

Comment:
For null, true, false, it's much safer to use asserTrue, assertNull etc. 
Replace and you may notice a failure